src/main/java/org/owasp/benchmark/score/parsers/SourceMeterReader.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * OWASP Benchmark Project
+ *
+ * This file is part of the Open Web Application Security Project (OWASP)
+ * Benchmark Project For details, please see
+ * <a href="https://www.owasp.org/index.php/Benchmark">https://www.owasp.org/index.php/Benchmark</a>.
+ *
+ * The OWASP Benchmark is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU General Public License as published by the Free Software Foundation, version 2.
+ *
+ * The OWASP Benchmark is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details
+ *
+ * @author Dave Wichers <a href="https://www.aspectsecurity.com">Aspect Security</a>
+ * @created 2015
+ */
+
+package org.owasp.benchmark.score.parsers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+
+public class SourceMeterReader extends Reader {
+
+
+    // Possible Cross-site Scripting Vulnerability:
+    // Source: String getValue()
+    // Sink: void println(String arg0)
+    // Trace:
+    // /home/istvan/owasp/ellenorzes/benchmark/src/main/java/org/owasp/benchmark/testcode/BenchmarkTest00023.java(81):(81,30,81,51)[0]
+    // /home/istvan/owasp/ellenorzes/benchmark/src/main/java/org/owasp/benchmark/testcode/BenchmarkTest00023.java(80):(80,33,81,51)[0]
+    // /home/istvan/owasp/ellenorzes/benchmark/src/main/java/org/owasp/benchmark/testcode/BenchmarkTest00023.java(80):(80,33,81,61)[0]
+    // /home/istvan/owasp/ellenorzes/benchmark/src/main/java/org/owasp/benchmark/testcode/BenchmarkTest00023.java(80):(80,4,81,62)[0]
+
+    public TestResults parse(File fileToParse) throws Exception {
+        TestResults tr = new TestResults("SourceMeter VulnerabilityHunter", true, TestResults.ToolType.SAST);
+
+        List<String> sourceLines = IOUtils.readLines(new FileInputStream(fileToParse));
+
+        String vuln = null;
+        String file = null;
+        boolean nextLine = false;
+        for (String line : sourceLines) {
+            try {
+                if (line.length() == 0) {
+                    vuln = null;
+                    file = null;
+                    nextLine = false;
+                }
+                if (line.startsWith("Possible ")) {
+                    System.out.println("\t" + line);
+                    vuln = line.substring("Possible ".length());
+                    vuln = vuln.substring(0, vuln.length() - " Vulnerability:".length() );
+                } else if (line.startsWith("Trace:")) {
+                    nextLine = true;
+                } else if (nextLine) {
+                    int idx = line.indexOf(".java");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 61 does not find the substring and returns -1, the call to substring() on line 62 will throw IndexOutOfBoundsException.